presto-hive/src/main/java/com/facebook/presto/hive/util/Statistics.java

```diff
@@ -115,4 +207,46 @@ public ReduceOperator flip()
             }
         }
     }
+
+    public static class Range
+    {
+        private final Optional<? extends Comparable<?>> min;
+        private final Optional<? extends Comparable<?>> max;
+
+        public static Range empty()
+        {
+            return Range.empty();
```

Comment:
is this endless recursion?

Reply:
And I've been wondering why is the test failing with the `StackOverflow` =))